docs/transformations/DeadCodeRemoval.md

```diff
@@ -0,0 +1,91 @@
+# Dead Code Removal
+
+## Overview
+
+The goal of this transformation is to remove all unnecessary code which is never executed during any call of the function. 
+
+## Example:
+
+```python
+def foo(x):
+    if x >= 0:
+        return x // 10
+        print(x)
+    return -x 
+    print(-x)
+```
+Transforms to:
+```python
+def foo(x):
+    if x >= 0:
+        return x // 10
+    return -x
+```
+
+## Algorithm:
+
+### Using Control Flow Graph
+1. Build [CFG](https://en.wikipedia.org/wiki/Control-flow_graph) for the program.
+2. Traverse the graph in reverse order and search for **unreachable** vertices.
+    The vertex is unreachable if: 
+    * The vertex is not the first in the graph.
+    * There is no incoming edges into this vertex.
+3. Delete the unreachable vertex. Iterate for all outcoming vertices from this vertex and again search for **unreachable** vertices.
+
+### Using Heuristics
+
+* For `while`, `if` evaluate conditions and remove the statements where condition is `False`.
```

Comment:
Maybe we should also consider 'False' condition for 'for'-loop? For example, when the list is empty.

Reply:
Is the empty list the only case for unreachable 'for'-loops?

Reply:
Sorry, didn't see your reply. I think yes, it's the only case and maybe it's not so important to implement, okay